sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/implementation/ConcurrencyLimitingSpliterator.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.monitor.ingestion.implementation;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+public class ConcurrencyLimitingSpliterator<T> implements Spliterator<T> {
```

Comment:
Add javadoc to explain how this works as this can potentially be moved to Azure Core.